src/config.h

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) 2017 Amaury SÉCHET
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CONFIG2_H
+#define BITCOIN_CONFIG2_H
+
+#include <boost/noncopyable.hpp>
+
+#include <cstdint>
+
+class CChainParams;
+
+class Config : public boost::noncopyable
+{
+public:
+    virtual const CChainParams &GetChainParams() const = 0;
+    virtual void SetCashAddrEncoding(bool) = 0;
+    virtual bool UseCashAddrEncoding() const = 0;
+};
+
+class GlobalConfig final : public Config
+{
+public:
+    GlobalConfig();
+    const CChainParams &GetChainParams() const;
+    void SetCashAddrEncoding(bool) override;
+    bool UseCashAddrEncoding() const override;
+
+private:
+    bool useCashAddr;
+};
+
+// Dummy for subclassing in unittests
+class DummyConfig : public Config {
+public:
+    const CChainParams &GetChainParams() const override;
+    void SetCashAddrEncoding(bool) override {}
+    bool UseCashAddrEncoding() const override { return false; }
+};
+
+// Temporary woraround.
```

Comment:
why is this a workaround, what should really be done?

Reply:
I didn't put the text there so I don't know why it was considered a workaround and I don't see how it could be different...

Reply:
Not my comment either, but I believe the intention is not have a global getter, but rather pass the global config instance as argument to everywhere it's needed.

Reply:
I'm ok with using a global since I can't conceive of a good reason for running multiple configs simultaneously.